MapboxMobileEvents/Categories/NSProcessInfo+SystemInfo.m

```diff
@@ -0,0 +1,65 @@
+#import "NSProcessInfo+SystemInfo.h"
+#include <sys/types.h>
+#include <sys/sysctl.h>
+#include <mach/machine.h>
+
+@implementation NSProcessInfo (SystemInfo)
+
++ (NSString*)stringForProcessorType:(NSInteger)type subtype:(NSInteger)subtype {
+
+    // Scratch String to build description. Ensures non-nil
+    NSMutableString *cpu = [[NSMutableString alloc] init];
+
+    // values for cputype and cpusubtype defined in mach/machine.h
+    if (type == CPU_TYPE_X86_64) {
+        [cpu appendString:@"x86_64"];
+    } else if (type == CPU_TYPE_X86) {
+        [cpu appendString:@"x86"];
+    } else if (type == CPU_TYPE_ARM) {
+        [cpu appendString:@"ARM"];
+        switch(subtype)
+        {
+            case CPU_SUBTYPE_ARM_V6:
+                [cpu appendString:@"V6"];
+                break;
+            case CPU_SUBTYPE_ARM_V7:
+                [cpu appendString:@"V7"];
+                break;
+            case CPU_SUBTYPE_ARM_V8:
+                [cpu appendString:@"V8"];
+                break;
+        }
+    } else if (type == CPU_TYPE_ARM64) {
+        [cpu appendString:@"ARM"];
+        switch(subtype)
+        {
+            case CPU_SUBTYPE_ARM64_V8:
+                [cpu appendString:@"V8"];
+                break;
+            case CPU_SUBTYPE_ARM64E:
+                [cpu appendString:@"E"];
+                break;
+        }
+    }
+
+    // Return Copy to Remove Mutability
+    return [cpu copy];
```

Comment:
Pretty sure that's just another mutable string, you want `[NSString stringWithString:cpu]` for an immutable copy

Reply:
If called upon NSMutable string it returns an immutable variant. If called with mutableString, it's responded with a mutable variant.

```
    // Return Copy to Remove Mutability
    NSMutableString * scratch = [cpu copy];
    [scratch appendString:@"Foo"]; >>> Results in exception
    
    return [cpu copy];
```
Due to:
`2020-06-17 15:53:31.032713-0700 MMETestHost[61435:5325632] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[NSTaggedPointerString appendString:]: unrecognized selector sent to instance 0xefb77832e6209c70'`

Although we could return a mutable string, this was just something I learned as a younger developer to prevent someone mucking with it later.

Reply:
Re-Reading, the copy isn't really necessary since we aren't retaining a reference. Removing it. Sorry for the confusion.

Reply:
> If called upon NSMutable string it returns an immutable variant

That's surprising, isn't it `instancetype`?